corehq/apps/hqwebapp/management/commands/resource_static.py

```diff
@@ -45,26 +30,9 @@ def output_resources(self, resources, overwrite=True, path=None):
             fout.write(yaml.dump([{'name': name, 'version': version}
```

Comment:
Am I reading this code correctly that resource_versions.yaml was always being overwritten? I believe that `open(path, 'w')` would have always overwritten it so the removal in commcare-cloud would have no affect. I think that if we actually needed to remove it, it would have been `'a'`.

I also think this may mean there's no dependency between this and the commacre-cloud PR

Reply:
Yes, `'w'` causes it to always be overwritten, but the previous resource versions are merged in when not overwriting:

https://github.com/dimagi/commcare-hq/blob/ea01bf98f49562baa130f48505ab361b85e93730/corehq/apps/hqwebapp/management/commands/resource_static.py#L22-L26

I'm assuming that `resource_versions.yaml` exists when `get_resource_versions(path=path)` is called; if it doesn't then that just returns an empty dict. I have not analyzed the places where that file is removed. Maybe it would make sense to have that fail hard if it does not exist but it is expected to exist?

Reply:
> I have not analyzed the places where that file is removed.

It's only removed by the deploy script, in the places affected by https://github.com/dimagi/commcare-cloud/pull/3222. My intent with this PR is to make these two management commands the **only** places that touch resource_versions.yaml.

I like the idea of erroring if resource_versions isn't found, added in 398c5e6239d393003dd60f9514abdac892066dc0  Given that I'm also removing the `rm`s, I think this is unlikely to trigger (though it would have made this bug much more apparent), but it won't hurt.